City-of-Helsinki/drupal-helfi-form-tool
# LOM-249: Auto logout extends and overrides

# [LOM-249](https://helsinkisolutionoffice.atlassian.net/browse/LOM-249)


## What was done


Refresh JWT token, if continuing session (dialog prompt)
Added module to extend & override some autologout logic without million patches.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout LOM-249-jwt-refresh`
  * `make fresh`
* Run `make drush-cr`

## How to test

* [ ] Helsinki profiili session should be refreshed when user continues session via dialog
    (easiest method to check if probably to set timeout to 1 min and hook to autologout_extend_page_attachments_alter with debugger and check that openid_expire updates after the dialog)
* [ ] Dialog should appear earlier / instantly during page load if helsinki profiili expiration is getting close to expiration.

## Other PRs
Helsinki Profiili.
https://github.com/City-of-Helsinki/drupal-module-helfi-helsinki-profiili/pull/8
